crates/ruff_linter/src/rules/pyupgrade/rules/printf_string_formatting.rs

```diff
@@ -434,14 +436,14 @@ pub(crate) fn printf_string_formatting(
     // Reconstruct the string.
     let mut contents = String::new();
     let mut prev_end = None;
-    for (range, format_string) in format_strings.into_iter() {
+    for (range, format_string) in format_strings {
         // Add the content before the string segment.
         match prev_end {
             None => {
                 contents.push_str(
                     checker
                         .locator()
-                        .slice(TextRange::new(string_expr.start(), range.start())),
+                        .slice(TextRange::new(bin_op.start(), range.start())),
```

Comment:
This should use the start of the entire binary expression instead of just the string expression. For example,

```py
  ("%d" "%d") % (1, 2)
# ^^^^^^^^^^^^^^^^^^^^ binary expression
#  ^^^^^^^^^ string expression
```